### PR TITLE
Add messaging component

### LINKED ISSUE #19: Chatroom skeleton function

### PR DESCRIPTION
**Pull request by rpollock03**
Resolves #19 #2 #21 - implement back and front end for messaging component with Firebase auth using stream chat SDKs

### CODE REVIEW

**Comment by piotrostr** [comment]
You need to add the environment variable of the messaging API key into the GitHub secrets

**Comment by rpollock03** [comment]
> You need to add the environment variable of the messaging API key into the GitHub secrets

Thanks @piotrostr done this.

File: components/Messaging.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ChannelList,
+  LoadingIndicator,
+} from "stream-chat-react"
+import "stream-chat-react/dist/css/index.css"
+
+import { getFunctions, httpsCallable } from "firebase/functions"
+import { useFirebaseApp } from "reactfire"
+import { useUser } from "reactfire"
+
+const apiKey = process.env.NEXT_PUBLIC_CHAT_API
```
Comment (piotrostr) [inline components/Messaging.tsx:20]: This variable is only used once, could drop the assingment

File: components/Messaging.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+const Messaging = () => {
+  const { status, data: currentUser } = useUser()
```
Comment (piotrostr) [inline components/Messaging.tsx:23]: The `status` variable has to be used to wait for the user to be retrieved

File: components/Messaging.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      const token = await getToken()
+      const streamToken = token.data.toString()
+
+      await chatClient.connectUser(
+        { id: currentUser.uid, name: currentUser.displayName },
+        streamToken
+      )
+
+      // To do: connect to front end, this will be the person you want to chat with
+      const recipient = {
```
Comment (piotrostr) [inline components/Messaging.tsx:46]: Move the `recipient` to props

File: components/Messaging.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        name: "Oli",
+      }
+
+      const channel = chatClient.channel("messaging", "chat-id", {
+        name: `${currentUser.displayName} and ${recipient.name}'s chat`,
```
Comment (piotrostr) [inline components/Messaging.tsx:51]: No need to include the display name of the current user

The current user will be shown in the header later

**Review by piotrostr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**